src/lib/Libifl/enc_reply.c

```diff
@@ -251,17 +251,21 @@ int
 encode_DIS_replyTPP(int sock, char *tppcmd_msgid, struct batch_reply *reply)
 {
 	int rc;
-	/* first encode "header" consisting of protocol type and version */
 
-	/* since it TPP connection, write a header */
-	if ((rc = is_compose(sock, IS_CMD_REPLY)) != DIS_SUCCESS)
+	/* first encode "header" consisting of protocol type and version */
+	if (reply->brp_choice == BATCH_REPLY_CHOICE_Status) {
+		if ((rc = ps_compose(sock, PS_STAT_RPLY)) != DIS_SUCCESS)
```

Comment:
u must be needing something like the tppcmd_msgid also for ps_compose()? If so, it is taken care of by ps_compose() itself?

Reply:
Presently we don't have a case where the sender will match the msgid with the reply for peer server messages.

Reply:
tpp_msgid is used in finding the right task to dispatch from the deferred task list. This is useful for commands like move job/run job to execute the associated post handler function. We may not have such a use case for stat requests at least in the near future.
See: process_DreplyTPP()